build/cmake/CMakeLists.txt

```diff
@@ -202,6 +209,30 @@ if(NOT LZ4_BUNDLED_MODE)
   install(FILES "${CMAKE_CURRENT_BINARY_DIR}/liblz4.pc"
     DESTINATION "${CMAKE_INSTALL_LIBDIR}/pkgconfig")
 
+  include(CMakePackageConfigHelpers)
+  write_basic_package_version_file(
+    "${CMAKE_CURRENT_BINARY_DIR}/lz4ConfigVersion.cmake"
+    VERSION ${LZ4_VERSION_STRING}
+    COMPATIBILITY SameMajorVersion)
+
+  set(LZ4_PKG_INSTALLDIR "${CMAKE_INSTALL_LIBDIR}/cmake/lz4")
+  configure_package_config_file(
+    "${CMAKE_CURRENT_LIST_DIR}/lz4Config.cmake.in"
+    "${CMAKE_CURRENT_BINARY_DIR}/lz4Config.cmake"
+    INSTALL_DESTINATION ${LZ4_PKG_INSTALLDIR})
+  export(EXPORT lz4Targets
+    FILE ${CMAKE_CURRENT_BINARY_DIR}/lz4Targets.cmake
+    NAMESPACE LZ4::)
```

Comment:
Why is the namespace upper case? The config file is called `lz4`, and exporting a namespace named differently is confusing, as it breaks the common practice used in the wider CMake ecosystem.

This also creates issues with pretty much every project who was already using lz4 in CMake, as people (quite unsurprisingly) use the lowercase `lz4` namespace, and package managers like [vcpkg do as well](https://github.com/microsoft/vcpkg/blob/42cc830d874f8e24eb4eb05bd5bb28496e3d498c/ports/lz4/CMakeLists.txt).